lib/repl.js

```diff
@@ -585,19 +595,20 @@ REPLServer.prototype.createContext = function() {
     context = global;
   } else {
     context = vm.createContext();
+    context.global = context;
+    context.global.global = context;
     const _console = new Console(this.outputStream);
-    Object.getOwnPropertyNames(global).forEach((name) => {
-      if (name === 'console') {
-        Object.defineProperty(context, name, {
-          configurable: true,
-          enumerable: true,
-          value: _console
-        });
-      }
-      else {
-        Object.defineProperty(context, name,
-          Object.getOwnPropertyDescriptor(global, name));
-      }
+    Object.defineProperty(context, 'console', {
+      configurable: true,
+      enumerable: true,
+      get: () => _console
+    });
+    Object.getOwnPropertyNames(global).filter((name) => {
+      if (name === 'console' || name === 'global') return false;
+      return GLOBAL_OBJECT_PROPERTY_MAP[name] === undefined;
+    }).forEach((name) => {
+      Object.defineProperty(context, name,
```

Comment:
I couldn't figure out the best way to test this programmatically, but the CLI behavior tests manually as expected.

    ~/s/node git:repl-useglobal-false-tab-completion ❯❯❯ ./node test-use-global-auto-complete.js
    > const x = new Int16Array()
    undefined
    > x instanceof Int16Array
    true
    >

I think this is the case, because there is no builtin shorthand for `Int16Array` and the like. For builtin objects that have a shorthand representation such as `[]` or `/\w+/` that representation is an `instanceof`the builtin language object no matter what. But there is no builtin shorthand for `Int16Array`.